internal/plugin/service/serviceintegration/userconfig.go

```diff
@@ -74,63 +73,63 @@ func flattenUserConfig(ctx context.Context, diags *diag.Diagnostics, o *resource
-	case isSet(o.DatadogUserConfig):
+	case schemautil.HasValue(o.DatadogUserConfig):
 		o.DatadogUserConfig = datadog.Flatten(ctx, diags, dto.UserConfig)
-	case isSet(o.ExternalAwsCloudwatchMetricsUserConfig):
+	case schemautil.HasValue(o.ExternalAwsCloudwatchMetricsUserConfig):
 		o.ExternalAwsCloudwatchMetricsUserConfig = externalawscloudwatchmetrics.Flatten(ctx, diags, dto.UserConfig)
-	case isSet(o.KafkaConnectUserConfig):
+	case schemautil.HasValue(o.KafkaConnectUserConfig):
 		o.KafkaConnectUserConfig = kafkaconnect.Flatten(ctx, diags, dto.UserConfig)
-	case isSet(o.KafkaLogsUserConfig):
+	case schemautil.HasValue(o.KafkaLogsUserConfig):
 		o.KafkaLogsUserConfig = kafkalogs.Flatten(ctx, diags, dto.UserConfig)
-	case isSet(o.KafkaMirrormakerUserConfig):
+	case schemautil.HasValue(o.KafkaMirrormakerUserConfig):
 		o.KafkaMirrormakerUserConfig = kafkamirrormaker.Flatten(ctx, diags, dto.UserConfig)
-	case isSet(o.LogsUserConfig):
+	case schemautil.HasValue(o.LogsUserConfig):
 		o.LogsUserConfig = logs.Flatten(ctx, diags, dto.UserConfig)
-	case isSet(o.MetricsUserConfig):
+	case schemautil.HasValue(o.MetricsUserConfig):
 		o.MetricsUserConfig = metrics.Flatten(ctx, diags, dto.UserConfig)
 	}
 }
 
 // expandUserConfig from terraform to aiven
-func expandUserConfig(ctx context.Context, diags *diag.Diagnostics, o *resourceModel, create bool) (map[string]any, error) {
-	var marshal func(any) (map[string]any, error)
-	if create {
-		marshal = schemautil.MarshalCreateUserConfig
-	} else {
-		marshal = schemautil.MarshalUpdateUserConfig
-	}
+func expandUserConfig(ctx context.Context, diags *diag.Diagnostics, o *resourceModel, create bool) map[string]any {
+	var config any
 
-	// If invalid integration type is set
+	// If an invalid integration type is set
 	// This will send wrong config to Aiven
 	// Which is sort of a validation too
 	switch {
-	case isSet(o.ClickhouseKafkaUserConfig):
-		return marshal(clickhousekafka.Expand(ctx, diags, o.ClickhouseKafkaUserConfig))
-	case isSet(o.ClickhousePostgresqlUserConfig):
-		return marshal(clickhousepostgresql.Expand(ctx, diags, o.ClickhousePostgresqlUserConfig))
-	case isSet(o.DatadogUserConfig):
-		return marshal(datadog.Expand(ctx, diags, o.DatadogUserConfig))
-	case isSet(o.ExternalAwsCloudwatchMetricsUserConfig):
-		return marshal(externalawscloudwatchmetrics.Expand(ctx, diags, o.ExternalAwsCloudwatchMetricsUserConfig))
-	case isSet(o.KafkaConnectUserConfig):
-		return marshal(kafkaconnect.Expand(ctx, diags, o.KafkaConnectUserConfig))
-	case isSet(o.KafkaLogsUserConfig):
-		return marshal(kafkalogs.Expand(ctx, diags, o.KafkaLogsUserConfig))
-	case isSet(o.KafkaMirrormakerUserConfig):
-		return marshal(kafkamirrormaker.Expand(ctx, diags, o.KafkaMirrormakerUserConfig))
-	case isSet(o.LogsUserConfig):
-		return marshal(logs.Expand(ctx, diags, o.LogsUserConfig))
-	case isSet(o.MetricsUserConfig):
-		return marshal(metrics.Expand(ctx, diags, o.MetricsUserConfig))
```

Comment:
`marshal` omitted errors that has been appended to `diags`. It should check for `HasError()` before marshal it. That's what PR does.